drizzlepac/haputils/quality_analysis.py

```diff
@@ -425,9 +446,257 @@ def generate_plots(json_data):
+        show(tabs)
+
+    
+    """
+    # TODO: include the date from the input data as part of the html filename
+    # Set the output file immediately as advised by Bokeh.
+    output_file(output)
+
+    # Setup the source of the data to be plotted so the axis variables can be
+    # referenced by column name in the Pandas dataframe
+    fitDF = ColumnDataSource(fit_data)
+    num_of_datasets = len(fit_data.index)
+    print('Number of datasets: {}'.format(num_of_datasets))
+    
+    colormap = [DETECTOR_LEGEND[x] for x in fitDF.data[HOVER_COLUMNS[1]]]
+    fitDF.data['colormap'] = colormap
+    inst_det = ["{}/{}".format(i,d) for (i,d) in zip(fitDF.data[HOVER_COLUMNS[0]], 
+                                         fitDF.data[HOVER_COLUMNS[1]])]
+    fitDF.data[INSTRUMENT_COLUMN] = inst_det
+    
+    plot_list = []
+
+    p1 = [build_circle_plot(x=RESULTS_COLUMNS[0], y=RESULTS_COLUMNS[1],
+                           source=fitDF,  
+                           title='RMS Values',
+                           x_label="RMS_X (pixels)",
+                           y_label='RMS_Y (pixels)',
+                           tips=[3, 0, 1, 2, 8],
+                           colormap=True, legend_group=INSTRUMENT_COLUMN)]
+    plot_list += p1
+                           
+    p2 = [build_circle_plot(x=RESULTS_COLUMNS[2], y=RESULTS_COLUMNS[3],
+                           source=fitDF,
+                           title='Offsets',
+                           x_label = "SHIFT X (pixels)",
+                           y_label = 'SHIFT Y (pixels)',
+                           tips=[3, 0, 1, 2, 8],
+                           colormap=True, legend_group=INSTRUMENT_COLUMN)]
+    plot_list += p2
+
+    p3 = [build_circle_plot(x=RESULTS_COLUMNS[8], y=RESULTS_COLUMNS[4],
+                           source=fitDF,
+                           title='Rotation',
+                           x_label = "Number of matched sources",
+                           y_label = 'Rotation (degrees)',
+                           tips=[3, 0, 1, 2, 8],
+                           colormap=True, legend_group=INSTRUMENT_COLUMN)]
+    plot_list += p3
+
+    p4 = [build_circle_plot(x=RESULTS_COLUMNS[8], y=RESULTS_COLUMNS[5],
+                           source=fitDF,
+                           title='Scale',
+                           x_label = "Number of matched sources",
+                           y_label = 'Scale',
+                           tips=[3, 0, 1, 2, 8],
+                           colormap=True, legend_group=INSTRUMENT_COLUMN)]
+    plot_list += p4
+    
+    p5 = [build_circle_plot(x=RESULTS_COLUMNS[8], y=RESULTS_COLUMNS[9],
+                           source=fitDF,
+                           title='Skew',
+                           x_label = "Number of matched sources",
+                           y_label = 'Skew (degrees)',
+                           tips=[3, 0, 1, 2, 8],
+                           colormap=True, legend_group=INSTRUMENT_COLUMN)]
+    plot_list += p5
+
```

Comment:
This generality is nice (!) for a gang of similar plots.

Reply:
Thanks.  This seemed like a nice way to organize all the plots that need to be generated on any given page.  Fortunately, Bokeh makes it very easy to do this!